### ISSUE TITLE
Macro bootstrap script expects zero argument constructor

### ISSUE THREAD

**Issue by TimWhiting** [dart-lang/sdk#48742]
@jakemac53 I was experimenting with the macro example on the language repo and ran into an error in the bootstrap script assigning to this map. It seems that there is some support for macro arguments maybe(?), but this made it so I could not experiment with them.
`Macro Function()` --> `Function` is probably how it has to change.
https://github.com/dart-lang/sdk/blob/6d543de174a229597184ee2404b7d53e328d731c/pkg/_fe_analyzer_shared/lib/src/macros/bootstrap.dart#L173

**Comment by jakemac53** [comment]
Yes, looks like that will need to change to just `Function`. The CFE also does not support arguments yet though (I know I saw a TODO in there recently).

**Comment by jakemac53** [comment]
Note that this code is also currently being modified, probably want to wait until https://dart-review.googlesource.com/c/sdk/+/239466 lands to change this just to avoid more conflicts

**Comment by TimWhiting** [comment]
> Note that this code is also currently being modified, probably want to wait until https://dart-review.googlesource.com/c/sdk/+/239466 lands to change this just to avoid more conflicts

I'll let you make any changes here. I don't have a dart-sdk set up for contributing right now. Just didn't know if you were aware of this.

**Comment by jakemac53** [comment]
I wasn't, although I am sure it would have come up once we tried to add the functionality to the CFE 👍 . But I can get ahead of things and fix it soon.

**Comment by devoncarew** [comment]
@jakemac53 - where should we track issues related to macros? In the language repo? Here, as an issue in `area-front-end`?
